src/core/chunked_stream.js

```diff
@@ -64,100 +74,43 @@ var ChunkedStream = (function ChunkedStreamClosure() {
       return chunks;
     },
 
-    getBaseStreams: function ChunkedStream_getBaseStreams() {
+    getBaseStreams: function ChunkedStreamBase_getBaseStreams() {
       return [this];
     },
 
-    allChunksLoaded: function ChunkedStream_allChunksLoaded() {
+    allChunksLoaded: function ChunkedStreamBase_allChunksLoaded() {
       return this.numChunksLoaded === this.numChunks;
     },
 
-    onReceiveData: function ChunkedStream_onReceiveData(begin, chunk) {
-      var end = begin + chunk.byteLength;
-
-      assert(begin % this.chunkSize === 0, 'Bad begin offset: ' + begin);
-      // Using this.length is inaccurate here since this.start can be moved
-      // See ChunkedStream.moveStart()
-      var length = this.bytes.length;
-      assert(end % this.chunkSize === 0 || end === length,
-             'Bad end offset: ' + end);
-
-      this.bytes.set(new Uint8Array(chunk), begin);
-      var chunkSize = this.chunkSize;
-      var beginChunk = Math.floor(begin / chunkSize);
-      var endChunk = Math.floor((end - 1) / chunkSize) + 1;
-      var curChunk;
-
-      for (curChunk = beginChunk; curChunk < endChunk; ++curChunk) {
-        if (!this.loadedChunks[curChunk]) {
-          this.loadedChunks[curChunk] = true;
-          ++this.numChunksLoaded;
-        }
-      }
-    },
-
-    onReceiveProgressiveData:
```

Comment:
keep `onReceiveData` and `onReceiveProgressiveData` is "abstract virtual": add them but throw error, e.g.  "Not implemented: ChunkedStreamBase_onReceiveData"